env0/resource_environment.go

```diff
@@ -929,17 +937,69 @@ func resourceEnvironmentDelete(ctx context.Context, d *schema.ResourceData, meta
 		return diag.Errorf(`must enable "force_destroy" safeguard in order to destroy`)
 	}
 
-	_, err := apiClient.EnvironmentDestroy(d.Id())
+	environment, err := apiClient.EnvironmentDestroy(d.Id())
 	if err != nil {
 		if frerr, ok := err.(*http.FailedResponseError); ok && frerr.BadRequest() {
 			tflog.Warn(ctx, "Could not delete environment. Already deleted?", map[string]interface{}{"id": d.Id(), "error": frerr.Error()})
 			return nil
 		}
 		return diag.Errorf("could not delete environment: %v", err)
 	}
+
+	if environment.Status != "INACTIVE" && d.Get("wait_for_destroy").(bool) {
+		if err := waitForEnvironmentDestroy(apiClient, &environment); err != nil {
+			return diag.FromErr(err)
+		}
+	}
+
 	return nil
 }
 
+func waitForEnvironmentDestroy(apiClient client.ApiClientInterface, environment *client.Environment) error {
+	waitInteval := time.Second * 10
+	timeout := time.Minute * 30
+	if os.Getenv("TF_ACC") == "1" { // For acceptance tests.
+		waitInteval = time.Second
+		timeout = time.Second * 10
+	}
+
+	ticker := time.NewTicker(waitInteval) // When invoked - check if environment is inactive.
+	timer := time.NewTimer(timeout)       // When invoked - time out
+	results := make(chan error)
+
+	go func() {
```

Comment:
this is some golang specific stuff (channels) not very common in other languages.
https://go.dev/tour/concurrency/2